feat: Exportación e importación del censo #4 #7

claugp06 opened the pull request:
Posibilidad de exportar el csv con los datos del censo

laurolmer committed:
feat: Exportación de censo a csv

laurolmer committed:
fix: Corrección de errores de export_to_csv()

Corrección de nombre de una variable que daba fallo.
Arreglo del formato CSV para que no se solape el nombre de las columnas

Commit:
tipo: Feat Export Census

Creación de una vista para la descarga del censo

Commit:
Plantilla HTML: Creación de la plantilla HTML

Se ha creadp la plantilla HTML para la feature import_csv

MarioArocaPaez reviewed (changes requested):
En principio, parece que todo está bien y debería funcionar correctamente, pero hay algunas cositas importantes.

Primero, es crucial destacar que no podemos hacer un PR a la rama master directamente desde una rama de feature. Primero debemos hacerlo a la rama develop y luego a master. :eyeglasses: :point_up_2:

Además, si no me equivoco, sería necesario añadir pruebas (tests) y creo que sería más lógico para la aplicación que solo los administradores puedan extraer el archivo CSV. Como ejemplo, podéis ver en el módulo de autenticación cómo se implementa para que solo un administrador pueda realizar esta operación

perealberto commented on decide/census/views.py:
```diff
@@ -7,13 +7,27 @@
 
+def export_to_csv(request):
+    census = Census.objects.all()
+    response = HttpResponse(
```
¡Está genial el código! Podría ser interesante utilizar `StreamingHttpResponse` en lugar de `HttpResponse` en caso de que el censo fuese muy grande para no tener problemas con el timeout.

Hay más información en la doc. oficial de django, en concreto en [esta página](https://docs.djangoproject.com/en/4.2/howto/outputting-csv/#streaming-large-csv-files).

Ya me decís qué pensáis :smile_cat:

perealberto reviewed (changes requested):
¡Ha quedado de lujo la vista!. Por lo demás, coincido con lo que ha comentado @MarioArocaPaez en la [review anterior](https://github.com/3ralon/decide/pull/10#pullrequestreview-1719756806).

laurolmer committed:
test: Adición de tests para export_to_csv()

El primer test prueba que el método devuelve los datos que están en el censo.
El segundo método prueba que cuando accede un usuario cualquiera devuelve un 200.

Commit:
Cambio nombre URL: Se ha cambiado la url de la descarga de csv

Se ha cambiado la direccion de descarga del csv para que sea mas fácil identificar

perealberto reviewed (commented):
Parece que todo está bien. ¿Vais a añadir más cambios?

laurolmer commented:
> Parece que todo está bien. ¿Vais a añadir más cambios?

No creo.

perealberto reviewed (approved):
De acuerdo, si los tests se pasan sin problemas y está hecho todo lo de esta feature, doy mi aprobado 😄 !
